packages/react-native-gesture-handler/src/v3/hooks/useGesture.ts

```diff
@@ -0,0 +1,116 @@
+import { useEffect, useMemo } from 'react';
+import { getNextHandlerTag } from '../../handlers/getNextHandlerTag';
+import RNGestureHandlerModule from '../../RNGestureHandlerModule';
+import { useGestureEvent } from './useGestureEvent';
+import { Reanimated } from '../../handlers/gestures/reanimatedWrapper';
+import { tagMessage } from '../../utils';
+
+type GestureType =
+  | 'TapGestureHandler'
+  | 'LongPressGestureHandler'
+  | 'PanGestureHandler'
+  | 'PinchGestureHandler'
+  | 'RotationGestureHandler'
+  | 'FlingGestureHandler'
+  | 'ForceTouchGestureHandler'
+  | 'ManualGestureHandler'
+  | 'NativeViewGestureHandler';
+
+type GestureEvents = {
+  onGestureHandlerStateChange: (event: any) => void;
+  onGestureHandlerEvent: (event: any) => void;
+  onGestureHandlerTouchEvent: (event: any) => void;
+  onGestureHandlerAnimatedEvent: (event: any) => void;
+};
+
+export interface NativeGesture {
+  tag: number;
+  name: GestureType;
+  config: Record<string, unknown>;
+  gestureEvents: GestureEvents;
+  shouldUseReanimated: boolean;
+  dispatchesAnimatedEvents: boolean;
+}
+
+function hasWorkletEventHandlers(config: Record<string, unknown>) {
+  return Object.values(config).some(
+    // @ts-ignore `__workletHash` comes from Reanimated and it does exist (on JS thread) if function is worklet.
+    (prop) => typeof prop === 'function' && prop.__workletHash !== undefined
```

Comment:
```suggestion
    (prop) => typeof prop === 'function' && '__workletHash' in prop
```

This seems to work, not sure if it introduces new edge-cases.

Reply:
> so it shouldn't be possible for __workletHash to be undefined

Well, if function is not marked as `worklet` it won't have `__workletHash` 🤷 

Also, I know it is not our use-case, but it might be worth mentioning that that these 2 [are not the same](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#description). Though I'm more used to `!== undefined`, I can see that this one looks better.

Reply:
>it might be worth mentioning that that these 2 [are not the same](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#description)

I know, that's why i added the "Note", to assure you that in this specific use-case, both `in` and `!== undef` will behave the same.

>Well, if function is not marked as worklet it won't have __workletHash 🤷

That's not oblivious at the first glance. Before I checked Reanimated source code, I considered a case where Reanimated would be setting `__workletHash` to `undefined` under some circumstances, which would make usage of `in` invalid.

Reply:
This is the exact case I was referring to in the "Note":
<img width="162" height="177" alt="image" src="https://github.com/user-attachments/assets/4a20b385-6260-430e-aa35-4b860a439310" />

Reply:
Done in [f9b8e1c](https://github.com/software-mansion/react-native-gesture-handler/pull/3617/commits/f9b8e1c198cdebfabfbbdef702f1de3876dcc9c5)